src/fmi4c.c

```diff
@@ -3706,70 +3700,100 @@ fmiHandle *fmi4c_loadFmu(const char *fmufile, const char* instanceName)
     getcwd(cwd, sizeof(char)*FILENAME_MAX);
 #endif
 
-    int argc = 6;
-    const char *argv[6];
-
+    // Decide location for where to unzip
+    char unzippLocation[FILENAME_MAX] = {0};
 #ifdef _WIN32
-    mkdir(instanceName);
+    DWORD len = GetTempPathA(FILENAME_MAX, unzippLocation);
+    if (len == 0) {
+       printf("Cant find temp path, using current directory\n");
+    }
+    // Create a unique tempfile and use its name for the unique directory name
+    char tempFileName[MAX_PATH];
+    GetTempFileNameA(unzippLocation, "", 0, tempFileName);
+
+    strncat(unzippLocation, "fmi4c_", FILENAME_MAX-strlen(unzippLocation)-1);
+    strncat(unzippLocation, instanceName, FILENAME_MAX-strlen(unzippLocation)-1);
+    strncat(unzippLocation, "_", FILENAME_MAX-strlen(unzippLocation)-1);
+    char * ds = strrchr(tempFileName, '\\');
+    if (ds) {
+        strncat(unzippLocation, ds+1, FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    else {
+        strncat(unzippLocation, tempFileName, FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    _mkdir(unzippLocation);
 #else
-    mkdir(instanceName, S_IRWXU | S_IRWXG | S_IRWXO);
+    const char* env_tmpdir = getenv("TMPDIR");
+    const char* env_tmp = getenv("TMP");
+    const char* env_temp = getenv("TEMP");
+    if (env_tmpdir) {
+        strncat(unzippLocation, env_tmpdir, FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    else if (env_tmp) {
+        strncat(unzippLocation, env_tmp, FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    else if (env_temp) {
+        strncat(unzippLocation, env_temp, FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    else if (access("/tmp/", W_OK) == 0) {
+        strncat(unzippLocation, "/tmp/", FILENAME_MAX-strlen(unzippLocation)-1);
+    }
+    // If no suitable temp directory is found, the current working directory will be used
```

Comment:
I assume this means that a temp directory is created inside the current working directory, and not that the files are unzipped directly inside the current directory?

Reply:
yes that is what I meant